src/embind/embind.js

```diff
@@ -892,11 +892,14 @@ var LibraryEmbind = {
     }
 
     if (returns) {
-        invokerFnBody += "var ret = retType.fromWireType(rv);\n" +
+        invokerFnBody += "var ret = retType.fromWireType(rv);\n";
 #if EMSCRIPTEN_TRACING
-                         "Module.emscripten_trace_exit_context();\n" +
+        invokerFnBody +=  "Module.emscripten_trace_exit_context();\n";
 #endif
-                         "return ret;\n";
+        if (deleteLater) {
+            invokerFnBody += "ret.deleteLater();"
```

Comment:
I think I understand why deleteLater() is necessary.  (Can you explain it to me in your own prose so I can verify if it matches my understanding?)

This seems pretty dangerous/risky.  It means that if someone were to hold onto the iterator across 'frames', then it would get deallocated.  The ES6 iteration protocol seems to have no mechanism for handling manual deallocation of iterator objects.

This makes me think this approach should be implemented another way somehow...  Like, it should delete the iterator when done iterating, but is there even a hook for that?  Ughhh JavaScript.  :|